src/nix/repl.cc

```diff
@@ -940,9 +937,20 @@ struct CmdRepl : InstallablesCommand
             auto installables = load();
             NixRepl::AnnotatedValues values;
             for (auto & installable: installables){
-                auto [val, pos] = installable->toValue(*state);
                 auto what = installable->what();
-                values.push_back( {val,what} );
+                if (!settings.isExperimentalFeatureEnabled(Xp::Flakes) && file){
```

Comment:
Don’t we want this regardless of the xp feature? It’d still be nice for `nix repl --file '<nixpkgs>'` to keep working

(maybe that could even be the case for the rest of the CLI which behaves a bit weirdly wrt autoargs)

Reply:
I can add it in, but there should be a consistent story across all of the CLI for the autoargs capability.

Reply:
Maybe yes. Let’s keep it as a follow-up for later